Added new container: diamond-aligner

Facundo1224 opened the pull request:
Changes:
- Added a new container: diamond-aligner.

Additional notes:
- Merged main into testing after pushing this change because I forgot to pull before adding the container.

Facundo1224 committed:
nuevo contenedor (diamond)

Facundo1224 committed:
Mergeando la branch 'main' con testing (me olvide de hacer pull antes de publicar la branch)

sbassi commented:
The code seems good, didn't test it yet. You need to move it into a directory named with the version name.
Instead of diamond-aligner/Dockerfile
should be:
diamond-aligner/2.1.9-debian/Dockerfile

sbassi commented on diamond-aligner/Dockerfile:
```diff
@@ -0,0 +1,33 @@
+    designed for high performance analysis of big sequence data" \
+    about.home="https://github.com/bbuchfink/diamond/" \
+    software.version="2.1.9-1" \
+    upstream.version="2.1.10" \
```
upstream should be 2.1.9

sbassi replied:
in upstream, you can use:
2.1.9.163
Since this is what the program reports.

sbassi commented on diamond-aligner/Dockerfile:
```diff
@@ -0,0 +1,33 @@
+
+WORKDIR /data
+
+ENTRYPOINT ["diamond"]
```
Remove Entrypoint and CMD.

sbassi commented on diamond-aligner/Dockerfile:
```diff
@@ -0,0 +1,33 @@
+USER root
+
+# 1 warning found LegacyKeyValueFormat: "ENV key=value" should be used instead of legacy "ENV key value" format (line 21)
+ENV DEBIAN_FRONTEND=noninteractive
```
replace this line as suggested in the warning and delete the warning comment

sbassi reviewed (changes requested):
The file is good, it works as is, but is better without the entry point, and there are minors changes required, so please address all comments.

Facundo1224 committed:
changes adjusted according to review

Facundo1224 commented:
Done! Changes in new commit:

- Adjusted file path to "diamond-aligner/2.1.9-debian/Dockerfile" (previously "diamond-aligner/Dockerfile")
- Corrected the "upstream.version" label to 2.1.9.163
- Removed the "ENTRYPOINT" and "CMD" layers from the Dockerfile.
- Removed leftover warning comment

sbassi committed:
fixes base image and ENV
